src/pages/Trade/components/OrderBookCard/MyTrades.tsx

```diff
@@ -1,50 +1,95 @@
-import React, { FC, useEffect } from 'react';
-import { connect, ConnectedProps } from 'react-redux';
-
-import {
-	fetchMyTradesRequest,
-	getMyTrades,
-	getIsLoadingMyTrades,
-	getIsLoadedMyTrades,
-} from 'ducks/trades/myTrades';
-
-import { RootState } from 'ducks/types';
-
-import useInterval from 'shared/hooks/useInterval';
+import React, { FC } from 'react';
+import { useQuery } from 'react-query';
+import snxData from 'synthetix-data';
+import { SYNTHS_MAP } from 'constants/currency';
 
 import TradeHistory from './TradeHistory';
 
 import { REFRESH_INTERVAL } from './constants';
+import { HistoricalTrades } from 'ducks/trades/types';
+import QUERY_KEYS from 'constants/queryKeys';
+import { CurrencyKey } from 'constants/currency';
 
-const mapStateToProps = (state: RootState) => ({
-	trades: getMyTrades(state),
-	isLoading: getIsLoadingMyTrades(state),
-	isLoaded: getIsLoadedMyTrades(state),
-});
-
-const mapDispatchToProps = {
-	fetchMyTradesRequest,
-};
+const MAX_TRADES = 100;
 
-const connector = connect(mapStateToProps, mapDispatchToProps);
+type MyTradesProps = {};
 
-type PropsFromRedux = ConnectedProps<typeof connector>;
+type SettledTrade = {
+	exchangeTimestamp: number;
+	dest: CurrencyKey;
+	src: CurrencyKey;
+	amount: number;
+	rebate: number;
+	reclaim: number;
+};
+type SettledTrades = SettledTrade[];
 
-type MyTradesProps = PropsFromRedux;
+const MyTrades: FC<MyTradesProps> = () => {
+	const currentWalletAddress = '0x2db6F5e838eD2BaD993E9FF2D3d7A5c1Cc35704C'; // TODO
+	const settledTradesQuery = useQuery<SettledTrades, any>(
+		QUERY_KEYS.HistoricalTrades.MyTrades(currentWalletAddress),
+		() =>
+			snxData.exchanger.exchangeEntriesSettled({
+				from: currentWalletAddress,
+				max: MAX_TRADES,
+			}),
+		{
+			refetchInterval: REFRESH_INTERVAL,
+		}
+	);
+	const tradesQuery = useQuery<HistoricalTrades, any>(
+		QUERY_KEYS.HistoricalTrades.MyTrades(currentWalletAddress),
+		() => {
+			const trades = snxData.exchanges.since({
+				fromAddress: currentWalletAddress,
+				maxBlock: Number.MAX_SAFE_INTEGER,
+				max: MAX_TRADES,
+			});
+			for (let i = 0; i < trades.length; i++) {
+				if (!!settledTradesQuery.data) {
```

Comment:
Am I allowed to mutate state inside of `react-query` before I return it or does this code need to be modified to be immutable?

Reply:
All of this `settledTradesQuery` logic is essentially duplicated from the `myTrades.js` duck.  Is it worth consolidating down to a single function?